### ISSUE TITLE
Issue with structs and : operator

### ISSUE THREAD

**Issue by TylerTheFox** [USCiLab/cereal#332]
I dont know what this operator does. It's in legacy code and I can't seem to find documentation on how it works.

```
    struct asdf 
    {
        int i : 4;
        template <class Archive>
        void serialize(Archive & ar)
        {
            ar(i);
        }
    };
```

Pretty much that won't compile.

cannot convert argument 2 from 'int' to 'int &'

**Comment by AzothAmmo** [comment]
It's a [bit field](https://en.wikipedia.org/wiki/Bit_field). I'll have to check if this is something we can easily support or whether it's somewhere we just make people work around it themselves.

**Comment by Brandantl** [comment]
Ah, I see it uses the first 4 bits, thanks. It's kind of hard to google syntax like that without knowing the name.

Now I haven't tested it but I assume something like this would probably work. It's kind of wasteful though.
`    struct asdf 
    {
        int i : 4;
        template <class Archive>
        void serialize(Archive & ar)
        {
            int i2 = i;
            ar(i2);
            i = i2;
        }
    };`

**Comment by erichkeane** [comment]
Non-const references to bit-fields are illegal by language spec.  You'll likely have to use the workaround mentioned in your last post.
